x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/Cast.java

```diff
@@ -78,24 +84,36 @@ protected TypeResolution resolveType() {
 
     @Override
     protected Processor makeProcessor() {
-        return new CastProcessor(SqlDataTypeConverter.converterFor(from(), to()));
+        return new CastProcessor(SqlDataTypeConverter.converterFor(from(), to()), zoneId);
     }
 
     @Override
     public ScriptTemplate asScript() {
         ScriptTemplate fieldAsScript = asScript(field());
-        return new ScriptTemplate(
-                formatTemplate(format("{sql}.", "cast({},{})", fieldAsScript.template())),
+
+        if (DataTypes.isDateTime(dataType)) {
```

Comment:
I'm not entirely sure what's better here: Only passing the timezone for datetime conversions results in a smaller diff of the changes and the resulting scripts are also not unnecessarily large. But it breaks a bit with consistency and might become a problem when more conversions start depending on the timezone.

Not sure whether anyone has a stronger opinion on this?